kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -114,11 +114,16 @@ private[kyo] class IOPromise[T](state: State[T])
             true
         }
 
+    private val nullCompletion = IOs(null)
+
     final def complete(v: T < IOs): Boolean =
+        val r =
+            if !isNull(v) then v
```

Comment:
Can you remind me why we need this function vs `ne null`?

Reply:
Ah right, thanks. I notice that function is using `==`... I'll sweep the code base later to switch all null checks to `eq/ne`

Reply:
Cool! I wonder if there's a way to configure a linter for that.